app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitTrackChairRatingTypesApiController.php

```diff
@@ -0,0 +1,231 @@
+use models\utils\IEntity;
+use ModelSerializers\SerializerRegistry;
+use utils\Filter;
+use utils\FilterElement;
+
+/**
+ * Class OAuth2SummitTrackChairRatingTypesApiController
+ * @package App\Http\Controllers
+ */
+final class OAuth2SummitTrackChairRatingTypesApiController
+    extends OAuth2ProtectedController
+{
+    use GetAndValidateJsonPayload;
+
+    use RequestProcessor;
+
+    /**
+     * @var ISummitRepository
+     */
+    private $summit_repository;
+
+    /**
+     * @var ISelectionPlanRepository
+     */
+    private $selection_plan_repository;
+
+    /**
+     * @var IPresentationTrackChairRatingTypeRepository
+     */
+    protected $repository;
+
+    /**
+     * @var ITrackChairRankingService
+     */
+    private $service;
+
+    /**
+     * OAuth2SummitTrackChairsApiController constructor.
+     * @param ISummitRepository $summit_repository
+     * @param ISelectionPlanRepository $selection_plan_repository
+     * @param IPresentationTrackChairRatingTypeRepository $repository
+     * @param ITrackChairRankingService $service
+     * @param IResourceServerContext $resource_server_context
+     */
+    public function __construct
+    (
+        ISummitRepository $summit_repository,
+        ISelectionPlanRepository $selection_plan_repository,
+        IPresentationTrackChairRatingTypeRepository $repository,
+        ITrackChairRankingService $service,
+        IResourceServerContext $resource_server_context
+    )
+    {
+        parent::__construct($resource_server_context);
+        $this->service = $service;
+        $this->summit_repository = $summit_repository;
+        $this->selection_plan_repository = $selection_plan_repository;
+        $this->repository = $repository;
+    }
+
+    use ParametrizedGetAll;
+
+    /**
+     * @return IBaseRepository
+     */
+    protected function getRepository(): IBaseRepository
+    {
+        return $this->repository;
+    }
+
+    /**
+     * @param $summit_id
+     * @param $selection_plan_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function getTrackChairRatingTypes($summit_id, $selection_plan_id) {
+
+        $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find(intval($summit_id));
+        if (is_null($summit)) return $this->error404();
+
+        $selection_plan = $this->selection_plan_repository->getById(intval($selection_plan_id));
+        if (is_null($selection_plan)) return $this->error404();
+
+        return $this->_getAll(
+            function () {
+                return [
+                    'selection_plan_id' => ['=='],
+                    'summit_id' => ['=='],
+                ];
+            },
+            function () {
+                return [
+                    'selection_plan_id' => 'sometimes|integer',
+                    'summit_id' => 'sometimes|integer',
+                ];
+            },
+            function () {
+                return [];
```

Comment:
we need an order per id and per order fields

Reply:
done